packages/node/src/indexer/blockDispatcher/worker-block-dispatcher.service.ts

```diff
@@ -71,6 +78,30 @@ async function createIndexerWorker(
         unfinalizedBlocksService.processUnfinalizedBlockHeader.bind(
           unfinalizedBlocksService,
         ),
+      hostAddToConnections:
+        connectionPoolState.addToConnections.bind(connectionPoolState),
+      hostGetNextConnectedApiIndex:
+        connectionPoolState.getNextConnectedApiIndex.bind(connectionPoolState),
+      hostGetFieldFromConnectionPoolItem:
+        connectionPoolState.getFieldValue.bind(connectionPoolState),
+      hostSetFieldInConnectionPoolItem:
+        connectionPoolState.setFieldValue.bind(connectionPoolState),
+      hostSetTimeoutIdInConnectionPoolItem:
+        connectionPoolState.setTimeout.bind(connectionPoolState),
+      hostClearTimeoutIdInConnectionPoolItem:
+        connectionPoolState.clearTimeout.bind(connectionPoolState),
+      hostGetSuspendedIndices:
+        connectionPoolState.getSuspendedIndices.bind(connectionPoolState),
+      hostDeleteFromPool:
+        connectionPoolState.deleteFromPool.bind(connectionPoolState),
+      hostHandleApiError:
+        connectionPoolState.handleApiError.bind(connectionPoolState),
+      hostHandleApiSuccess:
+        connectionPoolState.handleApiSuccess.bind(connectionPoolState),
+      hostGetDisconnectedIndices:
+        connectionPoolState.getDisconnectedIndices.bind(connectionPoolState),
+      hostShutdownPoolState:
+        connectionPoolState.shutdown.bind(connectionPoolState),
```

Comment:
Does a worker need access to all of these? It seems like the interface could be simplified

Reply:
we call all of these functions from `ConnectionPoolService`